pinot-common/src/main/java/org/apache/pinot/common/utils/DatabaseUtils.java

```diff
@@ -155,4 +157,18 @@ public static String extractDatabaseFromQueryRequest(
     String database = databaseFromHeaders != null ? databaseFromHeaders : databaseFromOptions;
     return Objects.requireNonNullElse(database, CommonConstants.DEFAULT_DATABASE);
   }
+
+  /**
+   * Extract the database name from the prefix of fully qualified table name.
+   * If no prefix is present "default" database is returned
+   */
+  public static String extractDatabaseFromFullyQualifiedTableName(String fullyQualifiedTableName) {
```

Comment:
looks like `fullyQualifiedTableName` is not necessarily a fully qualified table name, as there is check on split.length

if so, perhaps rename the method as extractDatabaseFromTableName()

Reply:
`fullyQualifiedTableName = <databaseName>.<logicalTableName>`
and we allow the databaseName to be skipped or have `default` value for backward compatibility hence table name without database prefix is also a valid `fullyQualifiedTableName`